packages/compass-import-export/src/components/import-modal.tsx

```diff
@@ -0,0 +1,290 @@
+  }, [startImport]);
+
+  // docsTotal is set to actual value only at the very end of processing a
+  // stream of documents
+  const isGuesstimated = useMemo(() => docsTotal === -1, [docsTotal]);
+
+  const importWasSuccessful = useMemo(
+    () => COMPLETED_STATUSES.includes(status),
+    [status]
+  );
+
+  return (
+    <Modal
+      open={isOpen}
+      setOpen={handleClose}
+      data-testid="import-modal"
+      trackingId="import_modal"
+    >
+      <ModalHeader title="Import" subtitle={`To Collection ${ns}`} />
+      <ModalBody>
+        <ImportOptions
+          delimiter={delimiter}
+          setDelimiter={setDelimiter}
+          fileType={fileType}
+          selectImportFileType={selectImportFileType}
+          fileName={fileName}
+          selectImportFileName={selectImportFileName}
+          stopOnErrors={stopOnErrors}
+          setStopOnErrors={setStopOnErrors}
+          ignoreBlanks={ignoreBlanks}
+          setIgnoreBlanks={setIgnoreBlanks}
+        />
+        {fileType === 'csv' && (
+          <ImportPreview
+            loaded={previewLoaded}
+            onFieldCheckedChanged={toggleIncludeField}
+            setFieldType={setFieldType}
+            values={values}
+            fields={fields}
+          />
+        )}
+        <ProgressBar
+          status={status}
+          withErrors={errors.length > 0}
+          cancel={cancelImport}
+          docsWritten={docsWritten}
+          docsProcessed={Math.max(docsProcessed, guesstimatedDocsProcessed)}
+          docsTotal={
+            // When guesstimating, guessed total might be too low, in that
+            // case the most reasonable thing to do would be to fallback to
+            // currently processed number
+            isGuesstimated
+              ? Math.max(docsProcessed, guesstimatedDocsTotal)
+              : docsTotal
+          }
+          progressLabel={(written: number, total: number) => {
+            return `${formatNumber(written)}\u00a0/\u00a0${
+              isGuesstimated ? '~' : ''
+            }${formatNumber(total)}`;
+          }}
+          progressTitle={(written: number, total: number) => {
+            return `Imported ${formatNumber(written)} out of ${
+              isGuesstimated ? 'approximately ' : ''
+            }${formatNumber(total)} documents`;
+          }}
+          message={MESSAGES[status]}
+        />
+        <ImportErrorList errors={errors} />
+      </ModalBody>
+      <ModalFooter>
+        {importWasSuccessful ? (
+          <Button
+            data-testid="done-button"
+            onClick={handleClose}
+            variant="primary"
+          >
+            Done
+          </Button>
+        ) : (
+          <>
+            <Button
+              data-testid="import-button"
+              onClick={handleImportBtnClicked}
+              disabled={!fileName || status === STARTED}
+              variant="primary"
+            >
+              {status === STARTED ? 'Importing...' : 'Import'}
```

Comment:
*obviously*, feel free to ignore:

```suggestion
              {status === STARTED ? 'Importing\u2026' : 'Import'}
```